### PR TITLE
Move to `ruff` for linting

### PR DESCRIPTION
**Pull request by samuelcolvin**

### CODE REVIEW

File: pydantic/main.py [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
     AnyClassMethod = classmethod[Any]
     TupleGenerator = typing.Generator[tuple[str, Any], None, None]
-    Model = typing.TypeVar('Model', bound='BaseModel')
+    Model = typing.TypeVar("Model", bound='BaseModel')
```
Comment (samuelcolvin) [inline pydantic/main.py:34]: @charliermarsh it looks like this incorrect quote example is not being picked up by ruff (at least for me locally).

Any idea what's going on?

Comment (charliermarsh) [reply pydantic/main.py:34]: Will take a look!

Comment (charliermarsh) [reply pydantic/main.py:34]: Ah, the `Q` codes aren't enabled by default -- only `E` and `F`.

You can change the `pyproject.toml` like this:

```
[tool.ruff]
line-length = 120
select = ["E", "F", "Q"]
flake8-quotes = {inline-quotes = 'single', multiline-quotes = 'double'}
```

Or, equivalently:

```
[tool.ruff]
line-length = 120
extend-select = ["Q"]
flake8-quotes = {inline-quotes = 'single', multiline-quotes = 'double'}
```

Comment (samuelcolvin) [reply pydantic/main.py:34]: that worked, thanks. Might be worth adding that to your example of `flake8-quotes`.

**Review by hramezani — APPROVED** [review]
LGTM :+1:

File: pydantic/main.py (hunk 1 above)
Comment (charliermarsh) [reply pydantic/main.py]: Good call. (Ruff needs proper docs.)

**Comment by charliermarsh** [comment]
Thanks @samuelcolvin and @hramezani! Ping whenever you have requests or run into issues. Happy to do custom work to support Pydantic as an early adopter!

**Pull request merged**